### PR TITLE
feat(eslint): disable function scoping

### PR DESCRIPTION
**Pull request by stevensacks**
This is a common pattern in React. The source of the measurements used as the basis for this rule were found to have been taken in the development version of React and were disproven. However, the myth persists. The React team has stated that this optimization is not necessary.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -116,6 +116,10 @@ module.exports = {
     // regular expressions.
     'unicorn/no-unsafe-regex': 'error',
 
+    // This is a common pattern in React, and has been proven to be a premature optimization
```
Comment (PixnBits) [inline index.js:119]: can you provide a link for the "proven" portion? the rule itself has two links to support it's claims on perf, it'd be good to provide a counterargument if/when this rule comes into discussion in the future 🙏

Comment (PixnBits) [reply index.js:119]: maybe a warning would be sufficient? probably depends on the analysis?

File: index.js [hunk 2]
```diff
@@ -116,6 +116,10 @@ module.exports = {
...
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/consistent-function-scoping.md
+    'unicorn/consistent-function-scoping': 'off',
```
Comment (JAdshead) [inline index.js:121]:
```suggestion
    'unicorn/consistent-function-scoping': 'warn',
```

File: index.js (hunk 1 above)
Comment (JAdshead) [inline index.js:119]:
```suggestion
```

Comment (JAdshead) [reply index.js:119]: I can't seem to find much supporting evidence either way

**Pull request merged**